src/metabase/models/setting.clj

```diff
@@ -443,28 +454,37 @@
   internally be `defsetting`; you shouldn't need to use it yourself."
   [{setting-name :name, setting-ns :namespace, setting-type :type, default :default, :as setting}]
-  (u/prog1 (let [setting-type         (s/validate Type (or setting-type :string))]
-             (merge
-              {:name        setting-name
-               :namespace   setting-ns
-               :description nil
-               :type        setting-type
-               :default     default
-               :on-change   nil
-               :getter      (partial (default-getter-for-type setting-type) setting-name)
-               :setter      (partial (default-setter-for-type setting-type) setting-name)
-               :tag         (default-tag-for-type setting-type)
-               :visibility  :admin
-               :sensitive?  false
-               :cache?      true}
-              (dissoc setting :name :type :default)))
-    (s/validate SettingDefinition <>)
-    ;; eastwood complains about (setting-name @registered-settings) for shadowing the function `setting-name`
-    (when-let [registered-setting (clojure.core/get @registered-settings setting-name)]
-      (when (not= setting-ns (:namespace registered-setting))
-        (throw (ex-info (tru "Setting {0} already registered in {1}" setting-name (:namespace registered-setting))
-                        {:existing-setting (dissoc registered-setting :on-change :getter :setter)}))))
-    (swap! registered-settings assoc setting-name <>)))
+  (let [munged-name (munge-setting-name (name setting-name))]
+    (u/prog1 (let [setting-type (s/validate Type (or setting-type :string))]
+               (merge
+                {:name        setting-name
+                 :munged-name munged-name
+                 :namespace   setting-ns
+                 :description nil
+                 :type        setting-type
+                 :default     default
+                 :on-change   nil
+                 :getter      (partial (default-getter-for-type setting-type) setting-name)
+                 :setter      (partial (default-setter-for-type setting-type) setting-name)
+                 :tag         (default-tag-for-type setting-type)
+                 :visibility  :admin
+                 :sensitive?  false
+                 :cache?      true}
+                (dissoc setting :name :type :default)))
+      (s/validate SettingDefinition <>)
+      ;; eastwood complains about (setting-name @registered-settings) for shadowing the function `setting-name`
+      (when-let [registered-setting (clojure.core/get @registered-settings setting-name)]
+        (when (not= setting-ns (:namespace registered-setting))
+          (throw (ex-info (tru "Setting {0} already registered in {1}" setting-name (:namespace registered-setting))
+                          {:existing-setting (dissoc registered-setting :on-change :getter :setter)}))))
+      (when-let [same-munge (first (filter (comp #{munged-name} :munged-name)
```

Comment:
if we don't like this linear scan we can add a lookup from munges to settings but it didn't feel necessary at our scale yet.